### ISSUE TITLE
Floating-base free-joint analytic IK Jacobian drifts with distance from world origin (regression from #2539)

### ISSUE THREAD

**Issue by dylanturpin** [newton-physics/newton#3866]
## Summary

In `newton.ik.IKJacobianType.ANALYTIC` mode, the inverse-kinematics Jacobian for a floating base (a root `JointType.FREE`, and equivalently `JointType.DISTANCE`) is wrong once the body is translated away from the world origin. The error in the analytic motion subspace grows linearly with the body's distance from the origin, so floating-base ANALYTIC IK drifts the farther the character/robot travels from the origin. In-place solves at the origin are unaffected, and `AUTODIFF` mode is correct throughout.

On a traveling motion clip we observed roughly **150 mm** of end-effector drift in ANALYTIC mode that does not appear in AUTODIFF mode.

## Affected versions

- Introduced in **1.4.0** by #2539 (commit `250fae60`).
- Still present in **1.5.0rc2** and on `main` (`1.6.0.dev0`).
- `AUTODIFF` mode is correct in all versions.

## Root cause

#2539 rewrote the FREE/DISTANCE analytic motion subspace (`write_free_distance_motion_subspace` in `newton/_src/sim/articulation.py`) so that the angular DOFs' rotation-to-translation coupling is taken about the **child body COM expressed in world coordinates**:

```python
x_child_com_world = wp.transform_point(X_wc, body_com_child)
...
joint_S_s[qd_start + 3] = wp.spatial_vector(-wp.cross(axis_world_x, x_child_com_world), axis_world_x)
```

For a floating base whose parent anchor is fixed, the origin-invariant coupling must be taken about the **fixed parent-joint anchor** (the translation of `X_pa_world`), not the traveling child COM. Because the child COM moves with the body, the `-cross(axis_world, x_child_com_world)` terms pick up a spurious contribution proportional to the body's world position, so the Jacobian's rotation-to-translation block is wrong by an amount that scales with distance from the origin. The public world-frame twist convention introduced by #2539 is otherwise correct and should be preserved.

## AUTODIFF-confirmed diagnosis

Comparing the ANALYTIC and AUTODIFF IK Jacobians for a single root FREE (or DISTANCE) body at increasing world translations shows exact agreement at the origin and a divergence that grows with translation. AUTODIFF matches finite differences at every translation; only ANALYTIC drifts. Equivalently, the analytic motion subspace `joint_S_s` for the root joint is not invariant under pure translation of the base, when it should be.

## Steps to reproduce

1. Build a model with a single root `FREE` joint (identity parent/child anchors).
2. Set the joint's translation to increasing world positions (0, 1, 5, 20 m) with an identity orientation.
3. Build two `IKSolver`s over an `IKObjectivePosition` on that body: one `AUTODIFF`, one `ANALYTIC`.
4. Compute each solver's Jacobian and compare. The max abs difference is ~0 at translation 0 and grows with translation under current `main`/1.5.0rc2.

## Minimal reproduction

```python
import numpy as np
import warp as wp
import newton
import newton.ik as ik

device = "cuda:0"

with wp.ScopedDevice(device):
    builder = newton.ModelBuilder()
    body = builder.add_link(mass=1.0)
    joint = builder.add_joint_free(
        parent=-1,
        child=body,
        parent_xform=wp.transform_identity(),
        child_xform=wp.transform_identity(),
    )
    builder.add_articulation([joint])
    model = builder.finalize(device=device, requires_grad=True)

    translations = (0.0, 1.0, 5.0, 20.0)
    joint_q = np.zeros((len(translations), model.joint_coord_count), dtype=np.float32)
    joint_q[:, 0] = translations   # translate the base along +x
    joint_q[:, 6] = 1.0            # identity quaternion (w = 1)

    targets = wp.zeros(len(translations), dtype=wp.vec3, device=device)
    objectives = [ik.IKObjectivePosition(body, wp.vec3(), targets)]

    solver_auto = ik.IKSolver(model, len(translations), objectives,
                              jacobian_mode=ik.IKJacobianType.AUTODIFF)
    solver_ana = ik.IKSolver(model, len(translations), objectives,
                             jacobian_mode=ik.IKJacobianType.ANALYTIC)

    q_auto = wp.array(joint_q, device=device, requires_grad=True)
    q_ana = wp.array(joint_q, device=device)
    solver_auto._impl._compute_residuals(q_auto)
    solver_ana._impl._compute_residuals(q_ana)
    J_auto = solver_auto._impl._jacobian_at(solver_auto._impl._ctx_solver(q_auto)).numpy()
    J_ana = solver_ana._impl._jacobian_at(solver_ana._impl._ctx_solver(q_ana)).numpy()

    for i, t in enumerate(translations):
        print(f"translation {t:>5} m: max|J_ana - J_auto| = {np.abs(J_ana[i] - J_auto[i]).max():.6e}")
```

Under current `main`/1.5.0rc2 the analytic-vs-autodiff error is ~0 at translation 0 and grows with translation (nonzero at 1/5/20 m). It should be 0 at every translation.

## Expected behavior

The ANALYTIC free/distance-joint IK Jacobian should equal the AUTODIFF Jacobian (and finite differences) regardless of the base's world position. The analytic motion subspace for a floating base must be origin-invariant: the angular DOFs' translation coupling should reference the fixed parent-joint anchor rather than the moving child COM.

**Comment by ganatrask** [comment]
**Independent reproduction on consumer Blackwell (sm_120), Newton 1.5.0 / Warp 1.16.0**

Confirmed, and the fault localises cleanly to the six floating-base Jacobian columns.

Setup: four end effectors with position + rotation targets plus joint limits (the objective
stack from `example_ik_h1.py`), LM, 32 iterations, sampler `NONE`. Targets are the forward
kinematics of a reference pose so they are exactly reachable; the solver is seeded from a small
perturbation of that pose (joints ±0.15 rad, base ±0.03 m/±0.05 rad, 8 seeds). The base is then
translated along +X and the same experiment repeated.

**The Jacobian itself.** Comparing `J_analytic` against `J_autodiff` at an identical
configuration, the disagreement in the six free-joint columns grows at **0.9999999 per metre**
of base distance — fitted over 2/5/10 m, on both Unitree H1 and Unitree R1 — while the
articulated columns are unchanged across the whole sweep (0.123614 on H1, 0.139031 on R1).

**The cause is identifiable directly.** For base angular DoF *j* the end-effector velocity
contribution is `cross(e_j, r)`; the two candidate moment arms are `r = p_ee` (about the world
origin) and `r = p_ee - p_base` (about the body). Comparing each mode's base angular columns
against both, at 10 m, max over all four end effectors:

| mode | vs `[p_ee]x` (world) | vs `[p_ee - p_base]x` (body) |
|---|--:|--:|
| H1 analytic | 10.0177 | **0.0452** |
| H1 autodiff | **9.5e-07** | 10.0193 |
| R1 analytic | 10.0285 | **0.0715** |
| R1 autodiff | **9.5e-07** | 10.0043 |

(9.5e-07 against terms of magnitude 10 is float32 round-off.)

So the **analytic** Jacobian takes the free joint's angular moment arm about the **body origin**,
while autodiff — which is the true derivative of Newton's own FK, and the mode that solves these
targets to 0.001 mm — takes it about the **world origin**. The difference between the two
candidate arms is exactly `[p_base]x`, which is why the disagreement is precisely proportional to
base distance rather than approximately so. (The nonzero floor at "distance 0", 1.0534 on H1 and
0.6987 on R1, is close to each robot's spawn pelvis height of 1.10 m and 0.74 m — consistent, but
~5 % off, so I would not lean on that step.)

**End-to-end, Unitree R1** (30 DoF), mean max-EE position error after 32 LM iterations:

| base distance | ANALYTIC | AUTODIFF | ANALYTIC, base pinned |
|--:|--:|--:|--:|
| 0 m | 0.357 mm | 0.001 mm | 0.001 mm |
| 1 m | 0.442 mm | 0.001 mm | 0.001 mm |
| 2 m | 0.541 mm | 0.001 mm | 0.001 mm |
| 5 m | 58.8 mm | 0.001 mm | 0.002 mm |
| 10 m | **139.0 mm** | 0.001 mm | 0.002 mm |

139 mm is right on the ~150 mm in the original report. It is not a convergence-budget problem:
the error is 139.013 mm at 16, 32 and 64 iterations alike.

**Control.** The right-hand column is the same ANALYTIC solve with the free joint's six DoFs
pinned through `joint_dof_mask`, seeded on the reference base pose. The end effectors are still
10 m from the origin and the error is flat at 0.002 mm. So this is not a general
distance-from-origin conditioning issue — it is those six columns.

**Worth flagging for anyone triaging with H1:** on H1 (51 DoF) the same defect only costs
~1.1 mm at 10 m at this perturbation size, because the extra redundancy plus LM's accept/reject
lets the solver route around a wrong descent direction. Increasing the seed perturbation to a
707 mm initial EE error brings H1 to 31.8 mm ANALYTIC vs 0.17 mm AUTODIFF (187×). A
lower-redundancy robot such as R1 exposes it immediately. An H1 run alone could easily read as
"minor".

The repro script and full JSON (per-distance for both robots, plus iteration and perturbation
sweeps, with GPU, arch and library versions recorded) are in a private repository — happy to share
the script or any specific numbers on request.

Everything needed to reproduce the Jacobian comparison independently is in the setup above, and it
is the cheap part: it needs no solver and no convergence luck. Build both candidate moment-arm
matrices for the free joint's angular DoFs at a fixed configuration, compare each mode's base
angular columns against both, and translate the base. The 1.0-per-metre slope falls out
immediately.
